Converters/CSi/Speckle.Converters.CSiShared/ToSpeckle/Raw/CSiFrameToSpeckleConverter.cs

```diff
@@ -0,0 +1,53 @@
+using Speckle.Converters.Common;
+
+namespace Speckle.Converters.CSiShared.ToSpeckle.Raw;
+
+/// <summary>
+/// Base converter for extracting frame properties common to all CSi applications.
+/// Product-specific implementations (ETABS, SAP2000) can extend this to add their own properties.
+/// </summary>
+public abstract class CSiFrameToSpeckleConverter
+{
+  protected CSiFrameToSpeckleConverter(IConverterSettingsStore<CSiConversionSettings> settingsStore)
+  {
+    SettingsStore = settingsStore;
+  }
+
+  protected IConverterSettingsStore<CSiConversionSettings> SettingsStore { get; }
+
+  /// <summary>
+  /// Extracts both common and product-specific properties from a frame element.
+  /// Product-specific properties are defined by derived classes.
+  /// </summary>
+  /// <remarks>
+  /// This will be refined! Just a POC for now. Data Extraction (Send) milestone will incorporate improvements here.
+  /// </remarks>
+  public Dictionary<string, object> GetClassProperties(CSiFrameWrapper frame)
+  {
+    var properties = new Dictionary<string, object>();
+
+    AddCommonClassProperties(frame, properties);
+    AddProductSpecificClassProperties(frame, properties);
+
+    return properties;
+  }
+
+  private void AddCommonClassProperties(CSiFrameWrapper frame, Dictionary<string, object> properties)
```

Comment:
i will let @clairekuang to judge mutating properties on abstract class, bc we were doing them with some extractors?

Reply:
For sure just a placeholder for now. We discussed for [CNX-882](https://linear.app/speckle/issue/CNX-882/send-etabsobjects-with-their-properties) how to do it better, so changes are coming on these definitions.